### PR TITLE
Update docker-credential-helpers dependency

### PR DESCRIPTION
**Pull request by vdemeester**
This is mainly for the `pass` helper ; `pass` shouldn't be called
every docker command anymore ;).

Linked to https://github.com/docker/docker-credential-helpers/issues/96

cc @euank

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: cli/config/credentials/default_store_linux.go [hunk 1]
```diff
@@ -5,7 +5,8 @@ import (
 )
 
 func defaultCredentialsStore() string {
-	if pass.PassInitialized {
+	pass := pass.Pass{}
```
Comment (silvin-lubecki) [inline cli/config/credentials/default_store_linux.go]: nit:
Too much pass 😵 
And not fan of shadowing a package by a variable.

Comment (n4ss) [reply cli/config/credentials/default_store_linux.go]: yeah same, besides that, LGTM!

**Comment by n4ss** [comment]
LGTM after a rename to passStore or something similar.

**Comment by vdemeester** [comment]
@n4ss @silvin-lubecki Updated :wink:

**Comment by thaJeztah** [comment]
@n4ss or @vdemeester should we tag a v0.6.1 release? https://github.com/docker/docker-credential-helpers/compare/v0.6.0...master

**Comment by thaJeztah** [comment]
Also; this fixes https://github.com/docker/cli/issues/699, correct?

**Comment by vdemeester** [comment]
@thaJeztah it fixes https://github.com/docker/docker-credential-helpers/issues/96 not #699 as it still execute *any* `pass` command in the `PATH`, just not for **all** docker commands.

**Comment by euank** [comment]
> pass shouldn't be called every docker command anymore ;).

I don't think that's correct and I don't think this actually fixes #699.

Before, `docker-credential-helpers` called pass in init so it was impossible for anyone to use the library without pass being called.

It's now possible to use it without pass running, but I don't think `docker/cli` does.

Specifically, the cli call here will execute pass any time it loads a config file which doesn't contain any auth configuration:

https://github.com/docker/cli/blob/ea65e9043c6ad3480882fd573c96a7cff4e66b8e/cli/config/config.go#L109-L120

This will allow people to avoid the call now by setting their config file in such a way that's skipped, but for a default empty config file it will still execute.

I think either credential checks need to be lazy such that it only initializes that if it's needed, not as soon as it loads the config file, or it needs to be a simpler initialization check there of just `LookPath` on the well-known 'pass' name.

**Comment by vdemeester** [comment]
@euank thanks for the comment :angel: I think you're right, somehow I though I also did sthg like https://github.com/docker/cli/pull/1131 in there :sweat_smile:

**Comment by euank** [comment]
I'd be happy to PR lazy initialization (something like https://github.com/euank/cli/commit/ae945b7ac52a2120960386e4ab7ba8ec908d7fd3) if that seems like a preferable approach... but I don't know if invoking `pass` on `docker pull` when no auth config is present is a reasonable behaviour or not.

Perhaps there should be `IsPresent` in `credential-helpers` and I was wrong to not include that.

Since I've realized lazy initialization will still end up invoking it even on `pull` of public images, it seems to me like either adding `IsPresent` to `credential-helpers` or directly doing `LookPath` in this codebase will be the best bet. Do you have any preference @vdemeester?

**Comment by vdemeester** [comment]
@euank I think you're right, either provide a `IsPresent` and using it in `docker/cli` or even simpler doing a `LookPath` in `docker/cli` seems fine by me
(cc @thaJeztah @n4ss )

**Comment by euank** [comment]
PR filed (#1160) @vdemeester

**Pull request merged**